parlai/chat_service/core/chat_service_manager.py

```diff
@@ -159,8 +160,10 @@ def _parse_config(self, opt):
         self.task_configs = self.config['configs']
         self.max_workers = self.config['max_workers']
         self.opt['task'] = self.config['task_name']
+        # Deepcopy the opts so the manager opts aren't changed by the world runner
```

Comment:
When the world runner calls `module_initialize` on a worlds file, it may add objects into the manager opts which get deep copied whenever a new agent is created. This causes infinite recursion if one of the objects added to the opts stores a reference to the same opts (ran into this issue and found this solution used internally)